beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -89,7 +91,9 @@ pub trait BeaconChainTypes: Send + Sync + 'static {
 pub struct BeaconChain<T: BeaconChainTypes> {
     pub spec: ChainSpec,
     /// Persistent storage for blocks, states, etc. Typically an on-disk store, such as LevelDB.
-    pub store: Arc<T::Store>,
+    pub store: Arc<RwLock<T::Store>>,
```

Comment:
As discussed, it would be nice for ergonomics if we could push the `RwLock` down into `HotColdDB`.

Reply:
Done!